### ISSUE TITLE
Audio/Video playback not synced with Multi-audio (HLS LIVE)

### ISSUE THREAD

**Issue by hiren3897** [shaka-project/shaka-player#4807]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
yes

**What version of Shaka Player are you using?**
4.3.0

**Can you reproduce the issue with our latest release version?**
No

**Can you reproduce the issue with the latest code from `main`?**
No

**Are you using the demo app or your own custom app?**
BOTH

**If custom app, can you reproduce the issue using our demo app?**
NA

**What browser and OS are you using?**
Chrome

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
NA

**What are the manifest and license server URIs?**

**What configuration are you using?  What is the output of `player.getConfiguration()`?**

manifest: {
      availabilityWindowOverride: Infinity,
    }

**What did you do?**

Tried to play the HLS live stream with 9 audio language

**What did you expect to happen?**
To play the HLS live stream smoothly with a lot of variants with sync in audio and video

**What actually happened?**

I see strange behavior in the playback of the HLS live stream
the audio is not synced with the video going on.

The audio is played almost 10s-16s early than the video.

I am not able to reproduce the issue can you please look into it please? it is urgent for us.
Is it because of the multi-audio? Lazy Loading

**Comment by avelad** [comment]
@hiren3897 Duplicate of https://github.com/shaka-project/shaka-player/issues/4599

**Comment by hiren3897** [comment]
Do we have any fixes on this since? or ETA?

**Comment by avelad** [comment]
@joeyparrish , can you help here?

**Comment by joeyparrish** [comment]
@theodab has identified a source of AV desync and is working on a fix.  Does the v4.2 branch work for you in the meantime?

**Comment by hiren3897** [comment]
Hello, thanks for your response

No I have the same problem with 4.2.x.
And we cannot use 4.2.x because it doesn’t have Lazy-Loading fixes. 

Because we have a live stream with 9 variants.

With due respect it could be good for us if we have that fix in 4.3.2

**Comment by joeyparrish** [comment]
Yes, we're working on it, and 4.3.2 is the next release.

**Comment by joeyparrish** [comment]
Can't access the content URL, so I'm closing this as a duplicate of #4599.  Please subscribe to that issue and confirm that the fix for that (when published) also fixes your issue.  If not, comment here with `@shaka-bot reopen`.  Thanks!

**Comment by joeyparrish** [comment]
Please see #4589, as well.  #4599 may be bad content.

**Comment by hiren3897** [comment]
@shaka-bot reopen

**Comment by hiren3897** [comment]
please find the test URL to see the problem: https://shaka-player-demo.appspot.com/demo/#audiolang=en-IN;textlang=en-IN;uilang=en-IN;asset=https://bhaktimarga-vod-p-cdn.hexaglobe.net/vod/202212261450/playlist.m3u8;panel=CUSTOM%20CONTENT;build=uncompiled

**Comment by joeyparrish** [comment]
@hiren3897, that isn't a live stream at all.  There's no PROGRAM-DATE-TIME and no DISCONTINUITY tags, and the content is completely static.  If audio is out of sync, it's either because of an issue with the segments themselves being fundamentally out of sync, or because of an issue with inaccurate durations in the playlists.

I see 4424 video segments and 4426 audio segments in the playlist, and a total duration of 2 hours 27 minutes.  But the audio is completely silent until after more than an hour, so you can't begin to judge AV sync until well into the presentation.  Around 1:06:51 is the first speech by anyone in front of the camera.

If the segment durations are inaccurate in the playlist, that's a long time for errors to accumulate as you sum the durations, and the player must use those summed durations to sync when seeking.  I could tell you for sure if it was a metadata problem by playing from 0, and waiting for over an hour to find out if audio would stay in sync without a seek, but that sounds like a waste of time to me.

The HLS solution for AV sync is PROGRAM-DATE-TIME tags, which are required for live streams.  This is an EVENT type stream, not actually live, so PROGRAM-DATE-TIME is not required by the spec here.  But if your playlist metadata isn't accurate enough, you'll need to add PROGRAM-DATE-TIME (PDT) tags to fix it.  Since PDT is wall-clock time, it doesn't accumulate rounding errors.

**Comment by hiren3897** [comment]
Thanks @joeyparrish 

I am sorry for providing the false URL 

According to the explanation I also found that the PROGRAM-DATE-TIME tags of audio and video are not synced in the live stream. also #EXT-X-MEDIA-SEQUENCE of audio in 2 more than video

so I suppose this is the problem of A/V sync PROGRAM-DATE-TIME tags

my team will try to sync  and see if we still have the bug

For example,
VIDEO playlist
```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-TARGETDURATION:6
#EXT-X-MEDIA-SEQUENCE:834
#EXTINF:6.000000,
#EXT-X-PROGRAM-DATE-TIME:2023-01-20T13:34:14.674+0000
livestreaming_video_360_20230120133406_000834_00595772_0000006000000.ts
#EXTINF:6.000000,
#EXT-X-PROGRAM-DATE-TIME:2023-01-20T13:34:20.674+0000
livestreaming_video_360_20230120133412_000835_00592952_0000006000000.ts
#EXTINF:6.120000,
#EXT-X-PROGRAM-DATE-TIME:2023-01-20T13:34:26.674+0000
livestreaming_video_360_20230120133418_000836_00595020_0000006120000.ts
#EXTINF:6.000000,
#EXT-X-PROGRAM-DATE-TIME:2023-01-20T13:34:32.794+0000
livestreaming_video_360_20230120133424_000837_00594832_0000006000000.ts
```

AUDIO Playlist
```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-TARGETDURATION:6
#EXT-X-MEDIA-SEQUENCE:836
#EXTINF:6.016000,
#EXT-X-PROGRAM-DATE-TIME:2023-01-20T13:34:25.357+0000 // ahead 10.683 seconds
livestreaming_audio_eng_20230120133407_000836_00006768_0000006016000.ts
#EXTINF:5.994667,
#EXT-X-PROGRAM-DATE-TIME:2023-01-20T13:34:31.373+0000 // ahead 10.699 seconds
livestreaming_audio_eng_20230120133413_000837_00006768_0000005994667.ts
#EXTINF:5.994667,
#EXT-X-PROGRAM-DATE-TIME:2023-01-20T13:34:37.368+0000 // ahead 10.694
livestreaming_audio_eng_20230120133419_000838_00006768_0000005994667.ts
#EXTINF:5.994667,
#EXT-X-PROGRAM-DATE-TIME:2023-01-20T13:34:43.362+0000 // ahead 10.568
livestreaming_audio_eng_20230120133425_000839_00006768_0000005994667.ts
```

**Comment by hiren3897** [comment]
@shaka-bot reopen

**Comment by hiren3897** [comment]
Hello @joeyparrish

I have emailed you my HLS live URL for test to reproduce the A/V sync problem.

I still have the problem.

I have added the PROGRAM-DATE-TIME tags in the playlist

**Comment by avelad** [comment]
@hiren3897 can you try with the main branch? Thanks!
